src/test/java/jewellery/inventory/integration/ProductCrudIntegrationTest.java

```diff
@@ -82,6 +83,42 @@ void setUp() {
         getProductRequestDto(Objects.requireNonNull(resourcesInUserResponseDto), user);
   }
 
+  @Test
+  void transferProductWithFakeId() {
+    UUID fakeId = UUID.randomUUID();
+
+    ResponseEntity<ProductResponseDto> response =
+        this.testRestTemplate.exchange(
+            getBaseProductUrl() + "/" + fakeId + "/transfer/" + fakeId,
+            HttpMethod.PUT,
+            null,
+            ProductResponseDto.class);
+
+    assertEquals(HttpStatus.NOT_FOUND, response.getStatusCode());
+  }
+
+  @Test
+  void transferProductCorrectData() {
+    ResponseEntity<ProductResponseDto> ProductResponse =
+        this.testRestTemplate.postForEntity(
+            getBaseProductUrl(), productRequestDto, ProductResponseDto.class);
+
+    ResponseEntity<UserResponseDto> userResponse =
+        this.testRestTemplate.postForEntity(
+            getBaseUserUrl(), userRequestDto, UserResponseDto.class);
```

Comment:
Can't we use the same pattern as the existing user instead of creating it through a request? 
`user = createUserInDatabase();`